### PR TITLE
make existing code Py3 compatible (WIP)

### PR DESCRIPTION
**Pull request by boegel**
This effectively makes `vsc-base` require Python 2.6, so be careful when merging this in (EasyBuild, which needs `vsc-base` will only move to Python 2.6 early 2015).

Currently WIP, not all unit tests pass yet with Python3 (but they still do using Python2.6).

Maybe this should include a bump to v2.0?

### CODE REVIEW

**Comment by boegel** [comment]
I still care about this, should I revive it and fix the remark,s @stdweird?

I hope to have Python 3 support in EB v3.

**Comment by stdweird** [comment]
we should wait after jens his talk.
@JensTimmerman you can look at this PR and add some examples how to move during the presentation?

**Comment by boegel** [comment]
OK, on hold until further notice then.

There are a couple of more involved things I couldn't figure out, so this PR is certainly not complete, but it a major step forward towards Py3 compatibility.

File: lib/vsc/utils/asyncprocess.py [hunk 1]
```diff
@@ -188,4 +190,9 @@ def send_all(p, data):
         sent = p.send(data)
         if sent is None:
             raise Exception(MESSAGE)
-        data = buffer(data, sent)
+        try:
+            if isinstance(data, str):
+                data = data.encode()
+            data = memoryview(data) #[:sent]  # sent?
+        except NameError:
+            data = buffer(data, sent)
```
Comment (JensTimmerman) [inline lib/vsc/utils/asyncprocess.py]: is this really what you intended?
`data` could be encoded here, or maybe not.

File: lib/vsc/utils/fancylogger.py [hunk 2]
```diff
@@ -72,6 +72,9 @@
 @author: Kenneth Hoste (Ghent University)
 """
 
+# required to avoid Py2 interpreting "print('a', 'b')" as printing a tuple
```
Comment (JensTimmerman) [inline lib/vsc/utils/fancylogger.py]: you want to reimplement six ?

File: lib/vsc/utils/frozendict.py [hunk 3]
```diff
@@ -17,14 +17,20 @@
...
-from UserDict import DictMixin
-
+import sys
+from distutils.version import LooseVersion
+pyver = sys.version.split(' ')[0]
+if LooseVersion(pyver) < LooseVersion('3.0'):
+    from UserDict import DictMixin as MutableMapping
+else:
+    # see http://stackoverflow.com/questions/11165188/how-to-achieve-the-functionality-of-userdict-dictmixin-in-python-3
+    from collections import MutableMapping
```
Comment (JensTimmerman) [inline lib/vsc/utils/frozendict.py]: this ugly check shouldn't be here at all, collections.MutableMapping is available in python 2.6, yes, everything is easier when you drop python 2.4 and 2.5 ;-)

File: lib/vsc/utils/generaloption.py [hunk 4]
```diff
@@ -50,6 +48,13 @@
 from vsc.utils.missing import shell_quote, nub
 from vsc.utils.optcomplete import autocomplete, CompleterOption
 
+pyver = sys.version.split(' ')[0]
+if LooseVersion(pyver) < LooseVersion('3.0'):
+    from ConfigParser import ConfigParser
+    from StringIO import StringIO
+else:
+    from configparser import ConfigParser
+    from io import StringIO
```
Comment (JensTimmerman) [inline lib/vsc/utils/generaloption.py]: `from io import StringIO` works fine in python 2.6

**Comment by boegel** [comment]
@JensTimmerman you were looking into an approach with `six`?

**Comment by boegel** [comment]
follow-up in #231

**Pull request closed without merging**